[cxx-interop] Name lookup failures in C++20 mode on Linux

Running `test/Interop/Cxx/concepts/method-requires.swift` on Linux exposes a name lookup issue:
```
test/Interop/Cxx/concepts/method-requires.swift:5:9: error: cannot find 'MyStruct' in scope
var s = MyStruct()
        ^~~~~~~~
```

This is only reproducible when `-Xcc -std=gnu++20` is passed.

--- a/lib/ClangImporter/ClangImporter.cpp
+++ b/lib/ClangImporter/ClangImporter.cpp
@@ -1255,6 +1255,9 @@ ClangImporter::create(ASTContext &ctx,
   // read them later.
   instance.getLangOpts().NeededByPCHOrCompilationUsesPCH = true;
 
+  // Clang implicitly enables this by default in C++20 mode.
+  instance.getLangOpts().ModulesLocalVisibility = false;
+
   if (importerOpts.Mode == ClangImporterOptions::Modes::PrecompiledModule)
     return importer;
 

--- a/test/Interop/Cxx/concepts/method-requires.swift
+++ b/test/Interop/Cxx/concepts/method-requires.swift
@@ -1,6 +1,6 @@
 // RUN: %target-swiftxx-frontend -emit-ir -Xcc -std=gnu++20 -I %S/Inputs %s | %FileCheck %s
 //
-// REQUIRES: OS=macosx
+// REQUIRES: OS=macosx || OS=linux-gnu
 
 import MethodRequires
 

--- a/test/Interop/Cxx/stdlib/use-std-vector.swift
+++ b/test/Interop/Cxx/stdlib/use-std-vector.swift
@@ -1,5 +1,5 @@
 // RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
-// FIXME: also test this in C++20 mode once rdar://108810356 is fixed.
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop -Xcc -std=gnu++20)
 //
 // REQUIRES: executable_test
 //

